packages/rainbowkit/src/config/getDefaultConfig.ts

```diff
@@ -1,15 +1,39 @@
-import { type Chain, configureChains, createConfig } from 'wagmi';
-import { publicProvider } from 'wagmi/providers/public';
-import type { WalletList } from '../wallets/Wallet';
+import { Transport } from 'viem';
+import {
+  CreateConfigParameters,
+  WagmiProviderProps,
+  createConfig,
+} from 'wagmi';
+import type { Chain } from 'wagmi/chains';
+import type {
+  RainbowKitWalletConnectParameters,
+  WalletList,
+} from '../wallets/Wallet';
 import { connectorsForWallets } from '../wallets/connectorsForWallets';
-import type { WalletConnectConnectorMetadata } from '../wallets/getWalletConnectConnector';
 import {
   coinbaseWallet,
   metaMaskWallet,
   rainbowWallet,
   walletConnectWallet,
 } from '../wallets/walletConnectors';
 
+type _chains = readonly [Chain, ...Chain[]];
+
+// Define the '_transports' type as a Record
+// It maps each 'Chain' id to a 'Transport'
+type _transports = Record<_chains[number]['id'], Transport>;
```

Comment:
Can we infer these types from Wagmi? I imagine they'd break if Wagmi fixes this strong readonly Chains typing

Reply:
I think they don't expose that type. They do it themselves here: 

![image](https://github.com/rainbow-me/rainbowkit/assets/53529533/170b6ac0-d3c0-4f40-baf2-64b30565d3ef)

Reply:
Yeah they would break. I think we can just a follow up PR on that if it happens.